src/api/friendships.js

```diff
@@ -0,0 +1,106 @@
+import createError from "http-errors";
+
+export default class API {
+  /**
+   * Bind service to router
+   *
+   * @param {Object} router the koa compatible router
+   */
+  bind(router) {
+    const listFriendshipsToId = async ctx => {
+      if (!ctx.params.userId) throw createError(400, "id in path is required.");
+
+      const req = {
+        userId: ctx.params.userId,
+        query: ctx.normalizedQuery || {},
+      };
+
+      const res = await this.listFriendshipsToId(req, ctx);
+
+      if (!res.body) throw createError(500, "should have body in response");
+
+      ctx.body = res.body;
+      ctx.status = 200;
+    };
+
+    const listFriendshipsFromId = async ctx => {
+      if (!ctx.params.userId) throw createError(400, "id in path is required.");
+
+      const req = {
+        userId: ctx.params.userId,
+        query: ctx.normalizedQuery || {},
+      };
+
+      const res = await this.listFriendshipsFromId(req, ctx);
+
+      if (!res.body) throw createError(500, "should have body in response");
```

Comment:
Do we need to return a 500 error?

Reply:
The API user won't get the error message here except the status code 500. But we create error in this way can give backend log better explanation about this error incident.